src/helpers/credential.rs

```diff
@@ -269,6 +260,142 @@ impl<'a> Credential<'a> {
         Ok(())
     }
 
+    pub async fn create_self_issued_credential(
+        &mut self,
+        schema_did: &str,
+        credential_subject_str: &str,
+        bbs_secret: &str,
+        bbs_private_key: &str,
+        credential_revocation_did: Option<&str>,
+        credential_revocation_id: Option<&str>,
+        exp_date: Option<&str>,
+    ) -> Result<String, CredentialError> {
+        let credential_subject: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let subject_did = credential_subject.id.unwrap_or("did:evan:default".to_string());
```

Comment:
as subject did doesn't seem to be an optional param , it is better to use ok_or instead of unwrap_or as ok_or will throw an error instead..
let subject_did = credential_subject.id.ok_or("no subject did found")?;

Reply:
Done, thanks